arabic/unpacked/tex.js

```diff
@@ -0,0 +1,205 @@
+
+        if (mapped !== text) {
+          token.data[0].data[0] = mapped;
+          token = token.With({
+            fontLang: 'ar'
+          });
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicIdentifier: function (token) {
+        var identifiersMap = MathJax.Hub.config.Arabic.identifiersMap;
+        var identifiersKeys = Object.keys(identifiersMap).sort(function (a, b) {
+          return b.length - a.length;
+        });
+
+        var text = token.data[0].data[0];
+        var mapped = text;
+
+        if ('chars' === token.data[0].type) {
+          // English Symbols like X and Y
+          identifiersKeys.forEach(function (enChar) {
+            var arChar = identifiersMap[enChar];
+            var regex = new RegExp(enChar, 'g');
+            mapped = mapped.replace(regex, arChar);
+          });
+        }
+
+        if (mapped !== text) {
+          token.data[0].data[0] = mapped;
+          token = token.With({
+            fontLang: 'ar'
+          });
+        }
+
+        return this.flipHorizontal(token);
+      },
+      arabicOperator: function (token) {
+        var operatorsMap = MathJax.Hub.config.Arabic.operatorsMap;
+        var text = token.data[0].data[0];
+        var mapped = text;
+
+        Object.keys(operatorsMap).forEach(function (enOperator) {
+          var regex = new RegExp('' + enOperator, 'g');
+          var arOperator = operatorsMap[enOperator];
+          mapped = mapped.replace(regex, arOperator);
+        });
+
+        if (mapped !== text) {
+          token = this.flipHorizontal(token).With({
+            fontLang: 'ar'
+          });
+
+          token.data[0].data[0] = mapped;
+        }
+
+        return token;
+      },
+      _getArgumentMML: function (name) {
+        //  returns an argument that is a single MathML element
+        //  (in an mrow if necessary)
+        //
+        //  This functions has been copied here from extensions/TeX/HTML.js, to avoid
+        //  adding a dependency.
+        //
+        //  TODO: Consider importing (as a dependency) this from HTML.js instead!
+        var arg = this.ParseArg(name);
+        if (arg.inferred && arg.data.length == 1)
+          {arg = arg.data[0]} else {delete arg.inferred}
+        return arg;
+      },
+      mmlToken: function (token) {
+        // TODO: Check for possible incompatibility with boldsymbol extension
+        var parsedToken = texParseMMLToken.apply(this, [token]);
+
+        if ('ar' === this.stack.env.lang) {
+          this.markArabicToken(parsedToken);
+        }
+
+        return parsedToken;
+      },
+      markArabicToken: function (token) {
+        if ('mn' === token.type) {
+          return this.arabicNumber(token);
+        } else if ('mi' === token.type) {
+          return this.arabicIdentifier(token);
+        } else if ('mo' === token.type) {
+          return this.arabicOperator(token);
+        }
+
+        return token;
+      },
+      AlignedArray: function () {
```

Comment:
There are also several similar array constructors in the AMSmath extension.  You may need to do something similar for them as well.